modules/images/webp-uploads/load.php

```diff
@@ -31,6 +31,11 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
 		return $output_format;
 	}
 
+	// Skip conversion when creating the `-scaled` image (for large image uploads).
+	if ( 1 === preg_match( '/.*-scaled\..{3}.?$/', $filename ) ) {
```

Comment:
Similar comment to @felixarntz would be ideal to check `!== false`  instead.

Additionally, the first part of the REGEX can be removed.

```suggestion
	if ( false !== preg_match( '/-scaled\..{3}.?$/', $filename ) ) {
```

Another approach as well can be using substr.

```php
if ( substr( $filename, -11, 7 ) === '-scaled' ) {
```

Reply:
i wanted to avoid substring which might make it harder to match files with .jpeg vs .jpg

Reply:
updated regex. return of false is a regex failure, 0 not found, 1 found, so i changed to a "truthy" check, i'll make sure that works